### DISCUSSION TITLE
What's the correct way of fetching thumbnails (avatars) of multiple channels? (cached?)

### DISCUSSION THREAD

**Discussion by tzwel** [FreeTubeApp/FreeTube#5408] [Q&A] [answered]
My question comes from [here](https://github.com/FreeTubeApp/FreeTube/issues/5211#issuecomment-2227145788)
I'm working on a feature that displays channel avatars below videos and I can't find a function/store that would do that easily in the codebase. What's the correct way to do this? Does FreeTube cache channels somewhere?
![image](https://github.com/user-attachments/assets/4a2b62d1-25ae-4e36-b14e-36dc97ca305a)

**Comment by absidue** [comment]
The correct way is to display them when they are available in the response from YouTube (e.g. search results) and otherwise not.

**Reply by tzwel** [reply]
Could you point me to the specific function that fetches YouTube search results or any other response you are talking about? Are you saying we shouldn't build some cache (if it doesn't exist, I don't know you haven't answered my question) to support this feature?
I don't know if I understand you well, you can clarify. I'm eager to implement some good qol features

**Reply by absidue** [reply, marked as answer]
Yes, I am saying don't build a cache, we don't do it for anything else, e.g. if the published date is missing then we just don't show it, same goes for the view count and duration and many other things.

The subscriptions page is the only one I would say that you should use the thumbnail URL from the profiles.

The easiest way to navigate around the code if you don't know it well is to start in the file for the page you are interested in. Each page in FreeTube has a subfolder in the `src/renderer/views` folder. When you are inside a JavaScript file and have found a function that you are interested in, most editors have a way to jump to that functions definition, in VSCode you can do that by CTRL + clicking on it.

**Reply by tzwel** [reply]
> The easiest way to navigate around the code if you don't know it well is to start in the file for the page you are interested in. Each page in FreeTube has a subfolder in the src/renderer/views folder

Figured this part myself, that's how I got to implement the feature for the subscribers tab!

> The subscriptions page is the only one I would say that you should use the thumbnail URL from the profiles.

Alright, since it already works for subs, I guess I can leave it like this and open the PR. Thanks!

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Closing as PR is opened so discussion can be continued over there.
